tests/unit/backend/test_tiledb.py

```diff
@@ -0,0 +1,273 @@
+                    },
+                ],
+                "is_preprint": True,
+                "journal": "Nature",
+                "published_day": 10,
+                "published_month": 10,
+                "published_year": 2022
+            }
+        self.db.edit_collection(id, "publisher_metadata", new_data)
+        # check
+        coll = self.db.get_collection(id)
+        self.assertEqual(coll['name'], "edited_name")
+        self.assertDictEqual(coll['publisher_metadata'], new_data)
+
+    def test_add_dataset_to_collection(self):
+        # Create collection
+        id = self.create_collection()
+        # Add dataset to collection
+        dataset_id = self.db.add_dataset(id, self.dataset_metadata)
+        # Get dataset/collection
+        datasets = self.db.get_datasets(id)
+        dataset_ids = self.db.get_attribute(id, "datasets")
+        self.assertEqual(datasets[0]['cell_count'], 12)
+        self.assertEqual(dataset_ids[0], dataset_id)
+
+    def test_delete_dataset_from_collection(self):
+        # Create collection
+        id = self.create_collection()
+        # Add dataset to collection
+        did1 = self.db.add_dataset(id, self.dataset_metadata)
+        did2 = self.db.add_dataset(id, self.dataset_metadata)
+        # Get datasets and delete one
+        self.db.delete_dataset(id, did1)
+        # Get dataset/collection
+        datasets = self.db.get_datasets(id)
+        self.assertEqual(datasets[0]['id'][0].decode("utf-8"), did2)
+
+    def test_collection_history(self):
+        # create collection
+        id = self.create_collection()
+        # edit collection
+        self.db.edit_collection(id, "name", "edited_name")
+        # read previous version
+        coll = self.db.read_collection_history(id, 1)  # 1 version back
+        self.assertEqual(coll['name'], "test_collection")
+        # revert to a previous version
+        self.db.revert_collection_history(id, 1)
+        coll = self.db.get_collection(id)
+        self.assertEqual(coll['name'], "test_collection")
+
+    def test_dataset_history(self):
+        # create collection
+        id = self.create_collection()
+        # add dataset
+        did = self.db.add_dataset(id, self.dataset_metadata)
+        # edit dataset
+        self.db.edit_dataset(did, "cell_count", 13)
+        # read previous version
+        dataset = self.db.read_dataset_history(did, 1)  # 1 version back
+        self.assertEqual(dataset['cell_count'], 12)
+        # revert to previous version
+        self.db.revert_dataset_history(did, 1)
+        dataset = self.db.get_dataset(did)
+        self.assertEqual(dataset['cell_count'], 12)
+
+    def test_create_revision_on_collection(self):
+        # Create a collection
+        id = self.create_collection()
+        # Create the revision
+        rev_id = self.db.create_revision(id)
+        # Get the revision
+        rev = self.db.get_collection(rev_id)
+        self.assertEqual(rev['name'], "test_collection")
```

Comment:
Add more assertions:
1. the revised collection is private
2. the revised collection id is != from the original collection id